tests/unit/s2n_client_hello_request_test.c

```diff
@@ -140,6 +195,35 @@ int main(int argc, char **argv)
         EXPECT_OK(s2n_test_send_and_recv(client_conn, server_conn));
     }
 
+    /* Test: no_renegotiation sent in response to valid hello request */
+    {
+        DEFER_CLEANUP(struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT),
+                s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(client_conn);
+        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, config));
+
+        DEFER_CLEANUP(struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER),
+                s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(server_conn);
+        EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config));
+
+        DEFER_CLEANUP(struct s2n_test_io_pair io_pair = { 0 }, s2n_io_pair_close);
+        EXPECT_SUCCESS(s2n_io_pair_init_non_blocking(&io_pair));
+        EXPECT_SUCCESS(s2n_connection_set_io_pair(client_conn, &io_pair));
+        EXPECT_SUCCESS(s2n_connection_set_io_pair(server_conn, &io_pair));
+
+        /* Complete the handshake */
+        EXPECT_SUCCESS(s2n_negotiate_test_server_and_client(server_conn, client_conn));
```

Comment:
can you check that the handshake is complete at this point

Reply:
same for the test above

Reply:
We're not testing s2n_negotiate_test_server_and_client here, so we trust that it negotiates the full handshake as it should. We don't check that the handshake completed everywhere that s2n_negotiate_test_server_and_client is used. If we did want some check other than "s2n_negotiate returned success for both client and server", then we could just as easily put that check in s2n_negotiate_test_server_and_client too.